app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/actions/AuthenticatedAction.scala

```diff
@@ -71,48 +78,82 @@ class AuthenticatedAction @Inject() (
       (selfBaseUrl + successRelativeUrl) -> (selfBaseUrl + failureRelativeUrl)
   }
 
-  @SuppressWarnings(Array("org.wartremover.warts.Product", "org.wartremover.warts.Serializable"))
-  override protected def refine[A](request: MessagesRequest[A]): Future[Either[Result, AuthenticatedRequest[A]]] = {
-    implicit val hc: HeaderCarrier = HeaderCarrierConverter.fromHeadersAndSession(request.headers, Some(request.session))
-
-    authorisedFunctions.authorised(ConfidenceLevel.L200).retrieve(Retrievals.nino) {
-      case None =>
-        logger.warn("Could not find NINO")
+  private def makeAuthenticatedRequestWithItmpRetrieval[A](
+    nino: String,
+    name: ItmpName,
+    dateOfBirth: LocalDate,
+    request: MessagesRequest[A]
+  ): Future[Either[Result, AuthenticatedRequest[A]]] =
+    name match {
+      case ItmpName(Some(givenName), _, Some(familyName)) =>
+        Right(AuthenticatedRequest(NINO(nino), Name(givenName, familyName), DateOfBirth(dateOfBirth), request))
+      case other =>
+        logger.warn(s"Failed to retrieve the complete name for this user: $other")
         Left(errorHandler.errorResult()(request))
+    }
 
-      case Some(nino) =>
-        Right(AuthenticatedRequest(NINO(nino), request))
-    }.recoverWith {
-      case _: NoActiveSession =>
-        Left(Redirect(signInUrl, Map("continue" -> Seq(selfBaseUrl + request.uri), "origin" -> Seq(origin))))
-
-      case _: InsufficientConfidenceLevel =>
-        handleInsufficientConfidenceLevel(request, errorHandler.errorResult()(request)).map(Left(_))
-
+  private def makeAuthenticatedRequestWithRetrieval[A](
+    nino: String,
+    name: RetrievalName,
+    dateOfBirth: LocalDate,
+    request: MessagesRequest[A]
+  ): Future[Either[Result, AuthenticatedRequest[A]]] =
+    name match {
+      case RetrievalName(Some(forename), Some(surname)) =>
+        Right(AuthenticatedRequest(NINO(nino), Name(forename, surname), DateOfBirth(dateOfBirth), request))
+      case other =>
+        logger.warn(s"Failed to retrieve the complete name for this user: $other")
+        Left(errorHandler.errorResult()(request))
     }
+
+  @SuppressWarnings(Array("org.wartremover.warts.Product", "org.wartremover.warts.Serializable"))
+  override protected def refine[A](request: MessagesRequest[A]): Future[Either[Result, AuthenticatedRequest[A]]] = {
+    implicit val hc: HeaderCarrier =
+      HeaderCarrierConverter.fromHeadersAndSession(request.headers, Some(request.session))
+
+    authorisedFunctions
+      .authorised(ConfidenceLevel.L200)
+      .retrieve(v2.Retrievals.nino and v2.Retrievals.itmpName and v2.Retrievals.name and Retrievals.itmpDateOfBirth) {
```

Comment:
could we not use `v2.Retrievals.itmpDateOfBirth` as well?

Reply:
yes, should have been v2.Retrievals.itmpDateOfBirth